packages/abi-typegen/src/templates/contract/factory.hbs

```diff
@@ -1,39 +1,29 @@
 {{header}}
 
-import { Interface, Contract, ContractFactory } from "fuels";
-import type { Provider, Account, AbstractAddress, BytesLike, DeployContractOptions, StorageSlot, DeployContractResult } from "fuels";
-import type { {{capitalizedName}}, {{capitalizedName}}Interface } from "../{{capitalizedName}}";
+import { ContractFactory } from "fuels";
+import type { Provider, Account, DeployContractOptions, DeployContractResult } from "fuels";
 
-const _abi = {{abiJsonString}};
+import { {{capitalizedName}} } from "./{{capitalizedName}}";
 
-const _storageSlots: StorageSlot[] = {{storageSlotsJsonString}};
+const bytecode = "{{hexlifiedBinString}}";
 
-export const {{capitalizedName}}__factory = {
-  abi: _abi,
+export class {{capitalizedName}}Factory extends ContractFactory {
 
-  storageSlots: _storageSlots,
+  static readonly bytecode = bytecode;
 
-  createInterface(): {{capitalizedName}}Interface {
-    return new Interface(_abi) as unknown as {{capitalizedName}}Interface
-  },
+  constructor(accountOrProvider: Account | Provider) {
+    super(bytecode, {{capitalizedName}}.abi, accountOrProvider);
```

Comment:
A note for `v1`:
If we make the `ContractFactory` take one param object as we do in `Predicate`, we could apply the same pattern of making optional the properties that are filled by typegen and typegend files wouldn't have a different constructor.

Reply:
Ties nicely with all my concerns around a more polished and intuitive API as well.

```ts
const { provider, wallet } = fuels(..);
const account = wallet(..);

cosnt bing = new MyPredicate({ account });
cosnt bang = new MyPredicate({ provider });
```

Or even:


```ts
fuels(FUEL_TEST_NETWORK)
  .wallet(WALLET_PVT_KEY)
  .predicate(MyPredicateFactory)
  .<xyz>();
```